cpp/tests/dictionary/remove_keys_test.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/copying.hpp>
+#include <cudf/dictionary/dictionary_column_view.hpp>
+#include <cudf/dictionary/encode.hpp>
+#include <cudf/dictionary/update_keys.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+#include <vector>
+
+struct DictionaryRemoveKeysTest : public cudf::test::BaseFixture {};
+
+TEST_F(DictionaryRemoveKeysTest, StringsColumn)
+{
+    cudf::test::strings_column_wrapper strings{ "eee", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "eee", "aaa" };
+    cudf::test::strings_column_wrapper del_keys{ "ddd", "bbb", "fff" };
+
+    auto dictionary = cudf::dictionary::encode( strings );
+    // remove keys
+    {
+        auto result = cudf::dictionary::remove_keys( cudf::dictionary_column_view(dictionary->view()), del_keys );
```

Comment:
General comment for this file, looks like there are quite a few declarations that could have `const` added to them
```suggestion
        auto const result = cudf::dictionary::remove_keys( cudf::dictionary_column_view(dictionary->view()), del_keys );
```